test/metabase/analytics/snowplow_test.clj

```diff
@@ -89,10 +89,12 @@
     (with-fake-snowplow-collector
       (snowplow/track-event! ::snowplow/new-instance-created)
       (is (= {:schema "iglu:com.metabase/instance/jsonschema/1-1-0",
-              :data {:id             (snowplow/analytics-uuid)
-                     :version        {:tag (:tag (public-settings/version))},
-                     :token_features (public-settings/token-features)
-                     :created_at     (snowplow/instance-creation)}}
+              :data {:id                           (snowplow/analytics-uuid)
+                     :version                      {:tag (:tag (public-settings/version))},
+                     :token_features               (public-settings/token-features)
+                     :created_at                   (snowplow/instance-creation)
+                     :application_database         (snowplow/app-db-type)
+                     :application_database_version (snowplow/app-db-version)}}
```

Comment:
I'm not convinced this is a great test since you could just have `app-db-version` return a hardcoded value like `noah_db_1.0` and it would still pass. `(is (= (x) (x))` is not a great way to test a function `x`... I'm trying to think of a better way to test this tho. I guess it gets tricky since we run different CI runs with different versions. Maybe changing this to a `schema=` could mean you could do something like a regex pattern for these values, which is better than nothing... or maybe since presumably there is only four possible values for `app-db-type` you could do an `s/enum`.

I guess it doesn't really matter that much -- maybe the complexity of the test outweighs the marginal benefits. If you manually check it and it looks good then that's probably ok for now. It's just analytics, and we can tweak it in the future if we need to.

Reply:
Yeah I agree that it doesn't test much, but I also couldn't think of a better way to test it consistently in all environments